core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/migration/MultiregionalBucketMigrationActor.scala

```diff
@@ -0,0 +1,857 @@
+          data = Map(
+            "migrationId" -> migration.id,
+            "workspace" -> workspace.toWorkspaceName
+          ) ++ data
+        )
+
+      for {
+        (storageService, userInfo, googleProjectToBill) <- asks { d =>
+          (d.storageService, d.userInfo, d.googleProjectToBill)
+        }
+
+        _ <- workspace.errorMessage.traverse_ { message =>
+          raiseError {
+            makeError("an error exists on workspace", Map("errorMessage" -> message))
+          }
+        }
+
+        workspaceBillingAccount = workspace.currentBillingAccountOnGoogleProject.getOrElse(
+          throw makeError("no billing account on workspace", Map.empty)
+        )
+
+        // Safe to assume that this exists if the workspace exists
+        billingProject <- inTransactionT { dataAccess =>
+          OptionT {
+            dataAccess.rawlsBillingProjectQuery.load(RawlsBillingProjectName(workspace.namespace))
+          }
+        }
+
+        billingProjectBillingAccount = billingProject.billingAccount.getOrElse(
+          throw makeError("no billing account on billing project", Map("billingProject" -> billingProject.projectName))
+        )
+
+        _ <- raiseWhen(billingProject.invalidBillingAccount) {
+          makeError(
+            "invalid billing account on billing project",
+            Map(
+              "billingProject" -> billingProject.projectName,
+              "billingProjectBillingAccount" -> billingProjectBillingAccount
+            )
+          )
+        }
+
+        _ <- raiseWhen(workspaceBillingAccount != billingProjectBillingAccount) {
+          makeError(
+            "billing account on workspace differs from billing account on billing project",
+            Map(
+              "workspaceBillingAccount" -> workspaceBillingAccount,
+              "billingProject" -> billingProject.projectName,
+              "billingProjectBillingAccount" -> billingProjectBillingAccount
+            )
+          )
+        }
+
+        requesterPaysEnabled <- liftIO {
+          for {
+            bucketOpt <- storageService.getBucket(
+              GoogleProject(workspace.googleProjectId.value),
+              GcsBucketName(workspace.bucketName),
+              bucketGetOptions = List(Storage.BucketGetOption.fields(Storage.BucketField.BILLING))
+            )
+
+            bucketInfo <- IO.fromOption(bucketOpt)(noWorkspaceBucketError(migration, workspace))
+            requesterPaysEnabled = Option(bucketInfo.requesterPays()).exists(_.booleanValue())
+
+            actorSaIdentity = serviceAccount(userInfo.userEmail.value)
+            _ <- storageService
+              .overrideIamPolicy(
+                GcsBucketName(workspace.bucketName),
+                Map(StorageRole.StorageAdmin -> NonEmptyList.one(actorSaIdentity)),
+                bucketSourceOptions =
+                  if (requesterPaysEnabled) List(BucketSourceOption.userProject(googleProjectToBill.value))
```

Comment:
Why does the user project need to be included here if it's requester pays? Does that get clobbered if omitted?

Reply:
If a bucket is requester pays, you always need to include the Google project to bill your operation to. The Google project to bill isn't needed (and may even be rejected, I'm not sure) if the bucket isn't requester pays.

In this case, we're billing to `broad-dsde-ENV` (which is `googleProjectToBill`) similar to how we're consuming `broad-dsde-ENV`'s quotas. There may be a yet-to-be-written ticket to move this and the quota over to use the user's project